### ISSUE TITLE
Loading css using <StyleSheet> crashes on iOS

### ISSUE THREAD

**Issue by lachlanwgordon** [xamarin/MobileBlazorBindings#140]
The sample weather App works on Android but when I run it on iOS I get a null reference error.

![Simulator Screen Shot - iPhone X 13 5 - 2020-07-21 at 15 05 26](https://user-images.githubusercontent.com/29908924/88015015-a04fb300-cb63-11ea-9e0b-a903731ebfe2.png)

I can however load in a css embeded resource using Assembly.GetManifestResource and add it to the applications resources, so the css file is definitely in the app correctly.

# Repro steps
1. Run MobileBlazorBindingsWeather.iOS on iPhone or Simulator
2. App never shows main page, just a white screen with a stack trace.

I'm currenlty building from source on latest Master but I get the same result on preview 3 nuget.

**Comment by Eilon** [comment]
I thought this looked super familiar. I think I debugged this like 6 months ago 😁 

The cause seems to be that the `ResourcesLoader` concrete service type isn't loaded. You cans see the dependency export here: https://github.com/xamarin/Xamarin.Forms/blob/719fc7a604ff0cce8922d717c99bfb0fa17e35e0/Xamarin.Forms.Xaml/ResourcesLoader.cs#L6

As to _why_ isn't the export being found, I'm still investigating. It's in the Xamarin.Forms assembly, which surely is loaded at this point, because that's where all the code is! But maybe something is causing the DependencyService to initialize _too early_, meaning that it already scanned the "loaded assembly list"? Very odd...

**Comment by Eilon** [comment]
OK I think I've figured it out. I keep forgetting that the Xamarin.Forms NuGet package contains _more than one assembly_!

The main DLLs it contains are:

* Xamarin.Forms.Core.dll
* Xamarin.Forms.Platform.dll
* Xamarin.Forms.Xaml.dll

Though there are dozens and dozens of other DLLs in platform-specific folders, as well as build infrastructure.

So I think all that needs to happen is to add a tiny bit of code that forces Xamarin.Forms.Xaml to get loaded.

**Comment by Eilon** [comment]
OK maybe a little _too_ familiar.

This was originally reported here: https://github.com/xamarin/MobileBlazorBindings/issues/109
And allegedly fixed here: https://github.com/xamarin/MobileBlazorBindings/commit/0bffddf6a04a016d99d94d3e4239bd4e417acf3d

Sadly, the fix done in that PR is exactly the fix I am proposing here, yet it clearly isn't working.

**Comment by Eilon** [comment]
I even tried forcing Xamarin.Forms.Xaml to load from the iOS startup code to ensure it was as early as possible, but it doesn't work either.

**Comment by lachlanwgordon** [comment]
Oops, I sent you down a rabbit hole.

Could it be something todo with the DependencyService not registering the Loader soon enough?

Out of interest, are you debugging this with Xamarin Forms built form Source? Or the Nuget? I tried stepping into the XF code with SourceLink but it took me to some seemingly unrelated code.

I'm be tempted to set up a solution with XF code and MBB Code together so I can step into anything, but I don't think my little virtual windows machine could hack it and I haven't managed to get MBB running on Mac.

**Comment by Eilon** [comment]
For this one I ran VS on my PC with the Mobile Blazor Bindings solution, running the iOS Simulator from PC on my Mac. For Xamarin.Forms I just looked at the source code on GitHub, matching lines up based on exception details 😁

**Comment by Eilon** [comment]
OK this should be fixed for real now in https://github.com/xamarin/MobileBlazorBindings/commit/c59c1d64ecaf83dde530b682cf56dbd46b560931

Turns out the change was quite simple: I had to make sure that I _ran_ arbitrary code in the Xamarin.Forms.Xaml assembly, not just referencing an arbitrary type in it. For some reason, on iOS just referencing a type doesn't necessarily cause the assembly to be loaded. This isn't necessarily surprising because code runs very differently on each platform.

So, this should now be fixed everywhere!
